File: src/frontend/src/types/cita.ts
export interface AgendarCitaDto {
  idCliente: number;
  idServicio: number;
  idTurno: number;
}

export interface CitaResponseDto {
  idCita: number;
  idCliente: number;
  clienteNombre: string;
  clienteCorreo: string;
  clienteTelefono: string;
  idServicio: number;
  servicioNombre: string;
  duracion: number;
  precio: number;
  idTurno: number;
  fecha: string;
  horaInicio: string;
  horaFin: string;
  idBarbero: number;
  barberoNombre: string;
  estado: string; // 'Pendiente', 'Atendida', 'Cancelada'
  fechaHora: string;
}

export interface AgendarCitaResponse {
  mensaje: string;
  cita: CitaResponseDto;
}

File: src/backend/BarberLosPeluchitos.Core/Entities/Cita.cs
namespace BarberLosPeluchitos.Core.Entities;

public class Cita
{
    public int IdCita { get; set; }
    public int IdCliente { get; set; }
    public int IdTurno { get; set; }
    public int IdServicio { get; set; }
    public DateTime FechaHora { get; set; } = DateTime.UtcNow;
    public string Estado { get; set; } = "Pendiente"; // 'Pendiente', 'Atendida', 'Cancelada'
    public int Duracion { get; set; }
    public decimal Precio { get; set; }

    // Relaciones de navegación
    public Cliente Cliente { get; set; } = null!;
    public Turno Turno { get; set; } = null!;
    public Servicio Servicio { get; set; } = null!;
}
